src/sentry/search/events/builder/metrics.py

```diff
@@ -592,7 +593,11 @@ def resolve_snql_function(
         alias: str,
         resolve_only: bool,
     ) -> SelectType | None:
-        if snql_function.snql_distribution is not None:
+        metric_id = arguments.get("metric_id")
```

Comment:
does `arguments.get("column")` work here to get the string mri? this prevents us from having to query the indexer (ie reverse_resolve) in the `_evaluate_metric_matches_function` later

Reply:
so can we do something like: 
1. if we find a valid prefix (ie d, s, g, c) and if the corresponding `snql_*` function is not defined, return early with a helpful error message
2. 
if snql_function.snql_distribution is not None and (if prefix = d or prefix is None):
  resolved_function = blah
if snql_function.snql_set is not None and (if prefix = s or prefix is None):
  resolved_function = blah
  
 so this will maintain existing behaviour and maybe more readable?

Reply:
It doesn't seem like `arguments.get("column")` returns the MRI, in my test case it's coming through as the column name `"mobile.slow_frames"`. The full payload for `arguments` looks like this:
```
{'column': 'mobile.slow_frames', 'if_col': 'release', 'if_val': 'foo', 'metric_id': 10016}
```

I guess this is because when we define the snql resolvers, we always use the `metric_id` to match for values and the other values are just the other args passed along. I could create a new `calculated_arg` to surface the MRI, but I don't think it's worth hoisting that logic up because it's used so deep in the query builder.